teps/0092-scheduling-timeout.md

```diff
@@ -17,361 +24,474 @@ authors:
   - [Non-Goals](#non-goals)
   - [Use Cases](#use-cases)
 - [Requirements](#requirements)
-    - [Boundaries of the timeout](#boundaries-of-the-timeout)
 - [Proposal](#proposal)
-  - [Notes/Caveats](#notescaveats)
-  - [Risks and Mitigations](#risks-and-mitigations)
-  - [User Experience](#user-experience)
-  - [Performance](#performance)
-- [Design Details](#design-details)
-  - [PipelineRun](#pipelinerun)
-  - [TaskRun](#taskrun)
-  - [What if the controller is totally overloaded?](#what-if-the-controller-is-totally-overloaded)
+  - [Scheduling Timeout](#scheduling-timeout)
+  - [Execution Timeout](#execution-timeout)
+  - [Total Timeout](#total-timeout)
+  - [API Changes](#api-changes)
+  - [Examples](#examples)
+  - [Validation](#validation)
+    - [Valid Cases](#valid-cases)
+    - [Invalid Cases](#invalid-cases)
   - [What if the pod gets rescheduled?](#what-if-the-pod-gets-rescheduled)
-  - [Syntax options](#syntax-options)
-    - [The new option(s)](#the-new-options)
-    - [The field in TaskRuns that represents the existing timeout](#the-field-in-taskruns-that-represents-the-existing-timeout)
-- [Test Plan](#test-plan)
+  - [PipelineRun](#pipelinerun)
+    - [TaskRunTemplate](#taskruntemplate)
+    - [TaskRunSpec](#taskrunspec)
+    - [TaskRunTemplate and TaskRunSpec](#taskruntemplate-and-taskrunspec)
+  - [Pipeline](#pipeline)
+  - [Notes and Caveats](#notes-and-caveats)
+    - [Pending State](#pending-state)
+    - [Timeouts in Kubernetes](#timeouts-in-kubernetes)
+  - [Risks and Mitigations](#risks-and-mitigations)
 - [Design Evaluation](#design-evaluation)
-- [Drawbacks](#drawbacks)
+    - [Reusability:](#reusability)
+    - [Simplicity](#simplicity)
+    - [Flexibility](#flexibility)
+    - [Conformance](#conformance)
+    - [Performance](#performance)
 - [Alternatives](#alternatives)
+  - [Configuration Levels](#configuration-levels)
+  - [Scheduling Timeout Boundaries](#scheduling-timeout-boundaries)
   - [Continue timeout until &quot;running&quot;](#continue-timeout-until-running)
   - [Leverage pending status](#leverage-pending-status)
-- [Infrastructure Needed](#infrastructure-needed)
+  - [Syntax Options](#syntax-options)
+  - [PipelineRun](#pipelinerun-1)
+- [Test Plan](#test-plan)
 - [Upgrade &amp; Migration Strategy](#upgrade--migration-strategy)
-- [Implementation Pull request(s)](#implementation-pull-requests)
 - [References](#references)
 <!-- /toc -->
 
 ## Summary
 
-Add runtime configuration options to allow setting a timeout for `TaskRuns` which applies before the `TaskRun's`
-underlying pod is scheduled, and only start our existing timeout counters after this timeout has passed, if specified.
+Add runtime configuration options to allow setting specific timeouts for parts of a `TaskRun` dedicated to scheduling the pod vs executing the pod.
+
+* The `scheduling` timeout applies from the start time of `TaskRun` until the scheduled time of the `Pod`.
```

Comment:
Do we mean "start time" in the status, or creation time ? I am asking this because, let's assume the controller is slow to pick up things, do we want to apply the timeout based of the creation time (when the user created the object) or when the controller first picked it up ?